bitcoin/src/crypto/key.rs

```diff
@@ -1841,3 +1872,33 @@ mod tests {
         assert!(PrivateKey::from_slice(&[1u8; 33], Network::Regtest).is_err());
     }
 }
+
+#[cfg(all(test, feature = "std"))]
+mod std_tests {
```

Comment:
This whole module is very unusual, can you explain what you are trying to achieve?